From ff3ab9d52d2d5bb9eed1cfcb54013d5483a37677 Mon Sep 17 00:00:00 2001
From: Craig Small <csmall@enc.com.au>
Date: Sun, 20 May 2012 17:08:29 +1000
Subject: [PATCH] watch compiles with 8bit enabled

watch wouldn't compile with --enable-watch8bit as some variables were
missing.  This changes fixes it.
---
 watch.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/watch.c b/watch.c
index 4465268..4649844 100644
--- a/watch.c
+++ b/watch.c
@@ -296,7 +296,11 @@ wint_t my_getwc(FILE * s)
 }
 #endif	/* WITH_WATCH8BIT */
 
+#ifdef WITH_WATCH8BIT
+void output_header(wchar_t *restrict wcommand, int wcommand_columns, int wcommand_characters, double interval)
+#else
 void output_header(char *restrict command, double interval)
+#endif	/* WITH_WATCH8BIT */
 {
 	time_t t = time(NULL);
 	char *ts = ctime(&t);
@@ -718,7 +722,11 @@ int main(int argc, char *argv[])
 		}
 
 		if (show_title)
+#ifdef WITH_WATCH8BIT
+			output_header(wcommand, wcommand_columns, wcommand_characters, interval);
+#else
 			output_header(command, interval);
+#endif	/* WITH_WATCH8BIT */
 
 		if (run_command(command, command_argv))
 			break;
-- 
1.7.8.6

